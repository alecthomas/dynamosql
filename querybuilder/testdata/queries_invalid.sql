-- Partition key required
SELECT * FROM gamescores
-- Partition key required
SELECT * FROM gamescores WHERE Wins = 3
-- $ numbered placeholder is not allowed
SELECT * FROM gamescores WHERE UserId = "101" AND Wins = $1
-- OR may not appear at the top level
SELECT * FROM gamescores WHERE UserId = :UserId OR Wins = 3
-- Partition key may not appear in a nested expression
SELECT * FROM gamescores WHERE UserId = :UserId AND (Wins = 3 OR UserId = "105")
-- Partition key may not appear in a NOT expression
SELECT * FROM gamescores WHERE NOT UserId = :UserId
-- Partition key must be in an equality condition
SELECT * FROM gamescores WHERE UserId > :UserId
-- Partition key must be in an equality condition
SELECT * FROM gamescores WHERE begins_with(UserId, 5)
-- Partition key may not appear twice
SELECT * FROM gamescores WHERE UserId = :UserId AND UserId = :UserId2
-- Sort key may not appear twice
SELECT * FROM gamescores WHERE UserId = :UserId AND GameTitle = "A" AND GameTitle = "B"
-- Sort key only be used with begins_with
SELECT * FROM gamescores WHERE UserId = :UserId AND attribute_exists(GameTitle)
