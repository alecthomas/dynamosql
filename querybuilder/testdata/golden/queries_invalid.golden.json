[
  {
    "Query": "SELECT * FROM gamescores",
    "Error": "partition key must appear exactly once in the WHERE clause, in an equality condition, such as: WHERE UserId = :param"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE Wins = 3",
    "Error": "partition key must appear exactly once in the WHERE clause, in an equality condition, such as: WHERE UserId = :param"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = \"101\" AND Wins = $1",
    "Error": "1:58: invalid token '$'"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = :UserId OR Wins = 3",
    "Error": "1:49: unexpected token \"OR\""
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = :UserId AND (Wins = 3 OR UserId = \"105\")",
    "Error": "partition key \"UserId\" may not appear in nested expression"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE NOT UserId = :UserId",
    "Error": "partition key must appear exactly once in the WHERE clause, in an equality condition, such as: WHERE UserId = :param"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId > :UserId",
    "Error": "partition key must appear exactly once in the WHERE clause, in an equality condition, such as: WHERE UserId = :param"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE begins_with(UserId, 5)",
    "Error": "partition key must appear exactly once in the WHERE clause, in an equality condition, such as: WHERE UserId = :param"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = :UserId AND UserId = :UserId2",
    "Error": "partition key \"UserId\" can only appear once in WHERE clause"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = :UserId AND GameTitle = \"A\" AND GameTitle = \"B\"",
    "Error": "sort key \"GameTitle\" can only appear once in WHERE clause"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = :UserId AND attribute_exists(GameTitle)",
    "Error": "sort key \"GameTitle\" may not be used with function attribute_exists()"
  },
  {
    "Query": "SELECT * FROM gamescores WHERE UserId = :UserId AND Wins = ?",
    "Error": "cannot mix positional params (?) with named params (:param)"
  }
]
